src/core/buffer/representation_buffer.ts

```diff
@@ -209,28 +178,174 @@ export default function RepresentationBuffer<T>({
   // Segments queued for download in the BufferQueue.
   let downloadQueue : IQueuedSegment[] = [];
 
+  // Emit when the current queue of download is finished
+  const finishedDownloadQueue$ = new Subject<void>();
+
   // Keep track of the informations about the pending Segment request.
   // null if no request is pending.
   let currentSegmentRequest : ISegmentRequestObject<T>|null = null;
 
   // Keep track of downloaded segments currently awaiting to be appended to the
   // SourceBuffer.
-  //
-  // This is to avoid scheduling another download for that segment.
-  // The ID of each segment (segment.id) is thus added before each append and
-  // removed after it.
   const sourceBufferWaitingQueue = new SimpleSet();
 
+  const status$ = observableCombineLatest(
+    clock$,
+    wantedBufferAhead$,
+    terminate$.pipe(take(1), mapTo(true), startWith(false)),
+    finishedDownloadQueue$.pipe(startWith(undefined))
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this redundant "undefined". ([typescript:S4623](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4623))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=344&issues=AWawYuC1LpqqSUlcRQ4h&open=AWawYuC1LpqqSUlcRQ4h)

[ISSUE_KEY]: AWawYuC1LpqqSUlcRQ4h